reference/reference_test.go

```diff
@@ -0,0 +1,62 @@
+package reference
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/docker/distribution/digest"
+)
+
+var refRegex = regexp.MustCompile(`^([a-z0-9]+(?:[-._][a-z0-9]+)*(?::[0-9]+(?:/[a-z0-9]+(?:[-._][a-z0-9]+)*)+|(?:/[a-z0-9]+(?:[-._][a-z0-9]+)*)+)?)(:[\w][\w.-]{0,127})?(@` + digest.DigestRegexp.String() + `)?$`)
+
+func getRepo(s string) string {
+	matches := refRegex.FindStringSubmatch(s)
+	if len(matches) == 0 {
+		return ""
+	}
+	return matches[1]
+}
+
+func testRepository(prefix string) error {
+	for _, s := range []string{
```

Comment:
I'd like to see a table of examples, similar to `names.go`. One can easily scan these examples see what is valid and what is not. Such a test will bolster the validity of this package.